addons/binding/org.openhab.binding.dwdunwetter/src/main/java/org/openhab/binding/dwdunwetter/internal/DwdUnwetterHandler.java

```diff
@@ -0,0 +1,224 @@
+            updateState(getChannelUuid(CHANNEL_ALTITUDE, i), data.getAltitude(i));
+            updateState(getChannelUuid(CHANNEL_CEILING, i), data.getCeiling(i));
+            if (warning == OnOffType.ON) {
+                updateState(getChannelUuid(DwdUnwetterBindingConstants.CHANNEL_WARNING, i), warning);
+            }
+            if (data.isNew(i)) {
+                triggerChannel(getChannelUuid(CHANNEL_UPDATED, i), "NEW");
+            }
+        }
+        data.updateCache();
+    }
+
+    /**
+     * Returns the configuration.
+     *
+     * @return The configuration
+     */
+    private DwdUnwetterConfiguration getDwdConfig() {
+        if (config == null) {
+            config = getConfigAs(DwdUnwetterConfiguration.class);
+        }
+        return config;
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Start initializing!");
+        updateStatus(ThingStatus.UNKNOWN);
+
+        data = new DwdWarningsData(getDwdConfig().getCellId());
+
+        List<Channel> channels = createChannels();
+        updateThing(editThing().withChannels(channels).build());
+
+        startAutomaticRefresh();
+
+        updateStatus(ThingStatus.ONLINE);
+        logger.debug("Finished initializing!");
+    }
+
+    private ChannelUID getChannelUuid(String typeId, int warningNumber) {
+        return new ChannelUID(getThing().getUID(), typeId + (warningNumber + 1));
+    }
+
+    private ChannelUID getChannelUuid(String typeId) {
+        return new ChannelUID(getThing().getUID(), typeId);
+    }
+
+    /**
+     * Creates a trigger Channel.
+     */
+    private Channel createTriggerChannel(String typeId, String label, int warningNumber) {
+        ChannelUID channelUID = getChannelUuid(typeId, warningNumber);
+        return ChannelBuilder.create(channelUID, "String") //
+                .withType(new ChannelTypeUID(DwdUnwetterBindingConstants.BINDING_ID, typeId)) //
+                .withLabel(label + " (" + (warningNumber + 1) + ")")//
+                .withKind(ChannelKind.TRIGGER) //
+                .build();
+    }
+
+    /**
+     * Creates a normal, state based, channel associated with a warning.
+     */
+    private Channel createChannel(String typeId, String itemType, String label, int warningNumber) {
+        ChannelUID channelUID = getChannelUuid(typeId, warningNumber);
+        return ChannelBuilder.create(channelUID, itemType) //
+                .withType(new ChannelTypeUID(DwdUnwetterBindingConstants.BINDING_ID, typeId)) //
+                .withLabel(label + " (" + (warningNumber + 1) + ")")//
+                .build();
+    }
+
+    /**
+     * Creates a normal, state based, channel not associated with a warning.
+     */
+    private Channel createChannel(String typeId, String itemType, String label) {
+        ChannelUID channelUID = getChannelUuid(typeId);
+        return ChannelBuilder.create(channelUID, itemType) //
```

Comment:
comment?

Reply:
What do you exactly mean? That method is commented.

Reply:
There is an empty comment at the end of the line. (Also for some of the other ones)

Reply:
They are there to keep the line breaks. I thing that using builders it's more readable

Reply:
Ok